app/controllers/concerns/account_concern.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module AccountConcern
+  extend ActiveSupport::Concern
+
+  ACCOUNT_SESSION_HEADER_INTERNAL_NAME = "HTTP_GOVUK_ACCOUNT_SESSION"
+  ACCOUNT_SESSION_HEADER_NAME = "GOVUK-Account-Session"
+  ACCOUNT_END_SESSION_HEADER_NAME = "GOVUK-Account-End-Session"
+  ACCOUNT_SESSION_DEV_COOKIE_NAME = "govuk_account_session"
+
+  included do
+    before_action :fetch_account_session_header
+    before_action :set_account_session_header
+    before_action :set_account_variant
+
+    helper_method :logged_in?
+
+    attr_accessor :account_session_header
+  end
+
+  def logged_in?
+    account_session_header.present?
+  end
+
+  def fetch_account_session_header
+    @account_session_header =
+      if request.headers[ACCOUNT_SESSION_HEADER_INTERNAL_NAME]
+        request.headers[ACCOUNT_SESSION_HEADER_INTERNAL_NAME]
+      elsif request.headers.to_h[ACCOUNT_SESSION_HEADER_NAME]
+        request.headers.to_h[ACCOUNT_SESSION_HEADER_NAME]
+      elsif Rails.env.development?
+        cookies[ACCOUNT_SESSION_DEV_COOKIE_NAME]
+      end
+  end
+
+  def show_signed_in_header?
+    account_session_header.present?
+  end
+
+  def set_account_variant
+    response.headers["Vary"] = [response.headers["Vary"], ACCOUNT_SESSION_HEADER_NAME].compact.join(", ")
+
+    set_slimmer_headers(
+      remove_search: true,
+      show_accounts: show_signed_in_header? ? "signed-in" : "signed-out",
+    )
+  end
```

Comment:
I think we should replace this with just setting no-cache headers for the sessions controller

Reply:
Is that `Cache-Control: no-cache`?

Reply:
Apparently [no-cache means that it can be cached, but has to be validated before being served](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control).  So I think we want a minimum of `no-cache, no-store`.

[This SO answer](https://stackoverflow.com/a/748646/1058828) suggests:

```
response.headers["Cache-Control"] = "no-cache, no-store"
response.headers["Pragma"] = "no-cache"
response.headers["Expires"] = "Mon, 01 Jan 1990 00:00:00 GMT"
```

Reply:
Yeah we were looking at that one too. Erred a bit as it was so old it was suggesting a `before_filter`.
But we'll stick that in the next commit for a try!